autotest/gcore/basic_test.py

```diff
@@ -756,3 +756,23 @@ def test_basic_test_DontUseExceptions():
     assert "ERROR " in err
     assert "FutureWarning: Neither gdal.UseExceptions()" not in err
     assert "FutureWarning: Neither ogr.UseExceptions()" not in err
+
+
+def test_create_context_manager(tmp_path):
+    numpy = pytest.importorskip("numpy")
+
+    fname = str(tmp_path / "out.tif")
+
+    drv = gdal.GetDriverByName("GTiff")
+    with drv.Create(fname, xsize=10, ysize=10, bands=1, eType=gdal.GDT_Float32) as ds:
+        dat = numpy.ones((10, 10), dtype=numpy.float32)
+        ds.GetRasterBand(1).WriteArray(dat)
+
+    # Make sure we don't crash when accessing ds after it has been closed
+    with pytest.raises(Exception):
+        ds.GetRasterBand(1).ReadAsArray()
+
+    ds_in = gdal.Open(fname)
+    dat_in = ds_in.GetRasterBand(1).ReadAsArray()
```

Comment:
yes I try to avoid using the numpy dependency as much as possible (admitedly this is a bit painful at times not using it)

you could actually use Fill(100) instead of Fill(0), and check the GetRasterBand(1).Checksum() value  (which should be != 0 with a fill value != 0)

Reply:
Got it, thanks